frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -1,14 +1,59 @@
 import React, { useContext, useState, useEffect } from 'react'
-import cx from 'classnames'
 import { updateAnnouncementVersion } from 'services/settings.service'
 
-import { PrimaryButton, ErrorBlock } from 'components/common'
 import { ModalContext } from 'contexts/modal.context'
 
-import styles from './AnnouncementModal.module.scss'
 import { i18n } from 'locales'
-import { ANNOUNCEMENT } from 'config'
-import { OutboundLink } from 'react-ga'
+import { ANNOUNCEMENT, getAnnouncementVersion } from 'config'
+import { ErrorBlock } from 'components/common'
+import GraphicAnnouncementModal from './GraphicAnnouncementModal'
+import VideoAnnouncementModal from './VideoAnnouncementModal'
+import styles from './AnnouncementModal.module.scss'
+
+export type Translation = string | undefined
+
+export type AnnouncementModalProps = {
+  title: Translation
+  mediaUrl: Translation
+  subtext: Translation
+  primaryButtonUrl: Translation
+  primaryButtonText: Translation
+  secondaryButtonUrl: Translation
+  secondaryButtonText: Translation
+  handleReadMoreClicked: () => Promise<void>
+}
+
+type Translations = Omit<AnnouncementModalProps, 'handleReadMoreClicked'>
+
+function getTranslations(): Translations {
```

Comment:
```suggestion
function getAnnouncementContent(): AnnouncementContent {
```

Consider using more specific names to help better convey intent/context for your code. See "Chapter 2: Meaningful Names" of [Clean Code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882).

Reply:
Thanks, that sounds a lot better! I've renamed these to `getAnnouncementContent()` and `AnnouncementContent`.